tests/components/fastdotcom/test_init.py

```diff
@@ -0,0 +1,104 @@
+"""Test for Sensibo component Init."""
+from __future__ import annotations
+
+from unittest.mock import patch
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant import config_entries
+from homeassistant.components.fastdotcom.const import DEFAULT_NAME, DOMAIN
+from homeassistant.const import EVENT_HOMEASSISTANT_STARTED
+from homeassistant.core import CoreState, HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_unload_entry(hass: HomeAssistant) -> None:
+    """Test unload an entry."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="UNIQUE_TEST_ID",
+        title=DEFAULT_NAME,
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fastdotcom.coordinator.fast_com", return_value=5.0
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert config_entry.state == config_entries.ConfigEntryState.LOADED
+    assert await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert config_entry.state is config_entries.ConfigEntryState.NOT_LOADED
+
+
+async def test_from_import(hass: HomeAssistant) -> None:
+    """Test imported entry."""
+    with patch(
+        "homeassistant.components.fastdotcom.coordinator.fast_com", return_value=5.0
+    ):
+        await async_setup_component(
+            hass,
+            DOMAIN,
+            {"fastdotcom": {}},
+        )
+        await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.fast_com_download")
+    assert state is not None
+    assert state.state == "5.0"
+
+
+async def test_delayed_speedtest_during_startup(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory
+) -> None:
+    """Test delayed speedtest during startup."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="UNIQUE_TEST_ID",
+        title=DEFAULT_NAME,
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fastdotcom.coordinator.fast_com", return_value=5.0
+    ), patch.object(hass, "state", CoreState.starting):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
```

Comment:
Could we check the `config_entry.state` here too? So we can check that the entry is only loaded after the `EVENT_HOMEASSISTANT_STARTED` event

Reply:
Config entry loads but it delays the coordinator to fetch data.
I extended the test and made a comment to explain this